test_simple_pe/test_simple_pe_magma.py

```diff
@@ -59,7 +67,141 @@ def test_simple_pe():
 
 
 """  # noqa
-    with open("test_simple_pe/build/problem.txt", "w") as f:
+    with open("test_simple_pe/build/problem_pe_core.txt", "w") as f:
         f.write(problem)
     assert not os.system(
-        "CoSA --problem test_simple_pe/build/problem.txt")
+        "CoSA --problem test_simple_pe/build/problem_pe_core.txt")
+
+
+@pytest.mark.parametrize("ops", ops)
+def test_simple_pe(ops):
+    pe = define_pe(ops, T=m.UInt, data_width=16)
+
+    class SimplePETester(ResetTester, ConfigurationTester):
+        pass
+
+    pe_functional_model = gen_simple_pe(ops, 16)()
+
+    tester = SimplePETester(pe, pe.clk, pe_functional_model)
+
+    m.compile("test_simple_pe/build/pe", pe, output="coreir",
+              coreir_args={"passes": ["rungenerators", "flatten",
+                                      "cullgraph"]})
+    # For some reason cullgraph above doesn't result in a culled output,
+    # perhaps coreir running it before rungenerators/flatten?
+    os.system("coreir -i test_simple_pe/build/pe.json -o "
+              "test_simple_pe/build/pe.json -p cullgraph")
+
+    # For verilator test
+    m.compile(f"test_simple_pe/build/{pe.name}", pe, output="coreir-verilog")
+
+    tester.zero_inputs()
+    opcode_width = m.bitutils.clog2(len(ops))
+    config_addr_width = 1
+
+    for config_data in [BitVector(x, opcode_width) for x in range(0, len(ops))]:
+        tester.reset()
+        tester.configure(BitVector(0, config_addr_width), config_data)
+        tester.test_vectors += \
+            generate_test_vectors_from_streams(
+                pe, pe_functional_model, {
+                    f"I{i}": lambda name, port: random_bv(len(port))
+                    for i in range(2)
+                })
+    tester.compile_and_run(directory="test_simple_pe/build",
+                           target="verilator", flags=["-Wno-fatal"])
+    opcode_width = m.bitutils.clog2(len(ops))
+    op_strs = {
+        operator.add: "+",
+        operator.sub: "-",
+        operator.and_: "&",
+        operator.or_: "|"
+    }
+
+    for i, op in enumerate(ops):
+        with open(f"test_simple_pe/build/conf_{op.__name__}.ets", "w") as ets:
+            ets.write(f"""\
+# INIT
```

Comment:
can we move this collateral elsewhere and read it in and use `.format()`? Kind of clunky inline.

Reply:
Moved to `common.configurable_circuit` as a string named `config_ets_template`

Reply:
Also abstracted the config_addr, config_data values, so it accepts the parameters `config_addr`, `config_addr_width`, `config_data`, and `config_data_width`.

Reply:
At some point it should be generalized to support multiple configuration address and data values.

I think we should build some abstraction for generating these configuration state machines. I think we can use the poke abstraction from fault to construct the sequence of signal values and compile it to this .ets format (it's similar to a test vector). For more complex state machines we can look into using silica or some other higher-level abstraction.